_signal_processing/krakenSDR_signal_processor.py

```diff
@@ -239,6 +241,46 @@ def resetPeakHold(self):
         if self.spectrum_fig_type == "Single":
             self.peak_hold_spectrum = np.ones(self.spectrum_window_size) * -200
 
+    def calculate_squelch(self, sampling_freq, N):
```

Comment:
@redradist It feels like this function must return earlier if auto squelch is not enabled.

Reply:
Unfortunately it cannot, because each channel could have option with automatic squelch

Reply:
Can you exit immediately if 
```
auto_squelch = any(is_enabled_auto_squelch(vfo_squelch_mode) for vfo_squelch_mode in self.vfo_squelch_mode)
```
is `False`?

Reply:
> Can you exit immediately if
> 
> ```
> auto_squelch = any(is_enabled_auto_squelch(vfo_squelch_mode) for vfo_squelch_mode in self.vfo_squelch_mode)
> ```
> 
> is `False`?

No, because in each channel could be set mode `Auto Channel`